tools/measure_coverage.sh

```diff
@@ -0,0 +1,138 @@
+    echo "                      the nodejs dependencies are installed."
+    echo ""
+    echo "The created html reports can be found in the 'coverage' directory,"
+    echo "which will be created in the IoT.js project source dir. The C and"
+    echo "JavaScript coverage reports are in the 'c' and 'js' subdirectories"
+    echo "respectively. The reports can be viewed by opening the 'index.html'"
+    echo "file in a web browser of your choice."
+    echo ""
+    echo "Running the script will require some additional dependencies:"
+    echo "    sudo apt install npm lcov nodejs gcc-multilib"
+    echo "    npm install --save-dev babel-cli nyc babel-plugin-istanbul"
+    echo ""
+    echo "    Note: nodejs version > 4.0.0 is required"
+    echo "    On older distros, you'll have to update your nodejs package with"
+    echo "    'curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -'"
+    echo "    before installing node."
+    echo ""
+    exit 0
+fi
+
+tools_dir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+iotjs_root=$(readlink -f "$tools_dir/..")
+
+dpkg -l npm >> /dev/null 2>&1 && \
+dpkg -l lcov >> /dev/null 2>&1 && \
+dpkg -l gcc-multilib >> /dev/null 2>&1 && \
+dpkg -l nodejs >> /dev/null 2>&1 && \
+test "$(nodejs -v)" \> "v4.0.0"
+
+if [ "$?" -ne "0" ]; then
+    echo "Please install all required dependencies"
+    echo "    sudo apt install npm lcov nodejs gcc-multilib"
+    echo "    npm install --save-dev babel-cli nyc babel-plugin-istanbul"
+    echo ""
+    echo "Note: nodejs version > 4.0.0 is required"
+    echo "    On older distros, you'll have to update your nodejs package with"
+    echo "    'curl -sL https://deb.nodesource.com/setup_4.x | sudo -E bash -'"
+    echo "    before installing node."
+    echo ""
+    exit 1
+fi
+
+modules_dir=$(readlink -f "$(npm bin)/..")
+
+if [ "$#" -gt "0" ]; then
+    path=$(readlink -f $1)
+    if [ ! -d "$path" ] || [ $(basename "$path") != "node_modules" ]; then
+        echo "'$1' is not a node_modules directory"
+        exit 1
+    fi
+
+    test -e $path/.bin/nyc && \
+    test -e $path/.bin/babel
+
+    if [ "$?" -ne "0" ]; then
+        echo "The specified node_modules directory does not provide the"
+        echo "required dependencies, you may need to install them."
+        echo ""
+        echo "    npm install --save-dev babel-cli nyc babel-plugin-istanbul"
+        exit 1
+    fi
+
+    modules_dir="$path"
+else
+    test -e $modules_dir/.bin/nyc && \
+    test -e $modules_dir/.bin/babel
+
+    if [ "$?" -ne "0" ]; then
+        echo "Could not find the node_modules directory, please specify it"
+        echo "as an argument, or install the node dependencies if you"
+        echo "haven't already."
+        echo ""
+        echo "    npm install --save-dev babel-cli nyc babel-plugin-istanbul"
+        exit 1
+    fi
+fi
+
+cd $iotjs_root
+
+# Transpile JS files to provide line counters
+/usr/bin/nodejs $modules_dir/.bin/babel \
```

Comment:
Are you sure this node is correct? You tested node from path for ver > 4.0. I could have old node installed in /usr/bin and new elsewhere with path pointing to it.